x-pack/plugins/security_solution/public/detection_engine/rule_management/api/hooks/use_find_rules_query.ts

```diff
@@ -54,6 +54,9 @@ export const useFindRulesQuery = (
     },
     {
       ...DEFAULT_QUERY_OPTIONS,
+      // Mark this query as immediately stale helps to avoid problems related to filtering.
+      // e.g. enabled and disabled state filter require data update which happens at the backend side
+      staleTime: 0,
```

Comment:
Thanks for the fix, @maximpn. And appreciate the comment here. 
Could you please also add `staleTime: 0` to the `useFetchRuleByIdQuery` hook? So we'll be sure no stale data is displayed on the rule detail page.

Reply:
Sure